SIL.Core/Xml/XmlSerializationHelper.cs

```diff
@@ -273,6 +274,48 @@ public static bool SerializeToFile<T>(string filename, T data, string rootElemen
 			return false;
 		}
 
+		/// ------------------------------------------------------------------------------------
+		/// <summary>
+		/// Serialize an object directly to disk, avoiding the operating system cache.
+		/// 
+		/// This method is intended to prevent the problem with null files being generated
+		/// as reported here: LT-20651, LT-20333, LTB-3915, LTB-3916, and LTB-3917
+		/// The theory is that FieldWorks is closing and power is lost before the operating
+		/// system cache is written to disk. This method is intended to prevent that problem
+		/// by avoiding the cache by writing through the cache directly to disk.
+		///
+		/// The disadvantage of this method is that it is slow; so it should only be used to
+		/// write relatively small files that are not frequently written.
+		/// </summary>
+		/// <param name="data">Object to be serialized to the file.</param>
+		/// <param name="path">The full path (containing the file name and extension).</param>
+		/// ------------------------------------------------------------------------------------
+		public static bool SerializeToFile_AvoidCache<T>(string path, T data, out Exception e)
```

Comment:
I think this is better. I do kind of wonder about the advisability of using a two-part name as opposed to adding another overload to SerializeToFile, with a Boolean parameter, But perhaps the value of that parameter would always be hard-coded, and the "false" case would always end up calling a simpler overload. Searching all of libpalaso, I do not see any precedence for a two-part name for a public API method. So I would suggest instead naming it SerializeToFileWithWriteThrough, or something like that.